lib/content/utils.js

```diff
@@ -69,6 +69,118 @@ const check = {
   },
 };
 
+/**
+ * Grab type from all content types in system
+ *
+ * @param {object} req - Express Request Object
+ *
+ * @returns {promise} reject with error message; resolve with single content-type object
+ */
+const type = (req) => {
+  return new Promise((res, rej) => {
+    const typ = req.params.type.toString();
+
+    if (!typ || typ === '') {
+      rej(config.content.messages.missing.type.replace('%type', '(no type given)'));
+    }
+
+    const exists = req.content.types.find(tp => {
+      return tp.id === typ.toLowerCase();
+    });
+
+    if (!exists || (exists === false) || (typeof exists !== 'object') || !exists.id || !exists.name) {
+      rej(config.content.messages.missing.type.replace('%type', typ));
+    }
+
+    res(exists);
+  });
+};
+
+/**
+ * Determine if an id is in uuid format
+ *
+ * @param {object} req - Express Request Object
+ *
+ * @returns {promise} reject with error message; resolve true
+ */
+const id = (req) => {
+  return new Promise((res, rej) => {
+    // no id in the request
+    if (!req.params.hasOwnProperty('id')) {
+      res(true);
+    }
+
+    // this can be the add, approve, or save action instead of a uuid
+    if (req.params.id === config.content.actions.add || req.params.id === config.content.actions.save || req.params.id === config.content.actions.approve) {
+      res(true);
+    }
+
+    if (req.params.id === '' || typeof req.params.id !== 'string' || !isUUID(req.params.id)) {
+      rej(config.content.messages.format.id);
+    }
+
+    res(true);
+  });
+};
+
+/**
+ * Determine if a revision is a number
+ *
+ * @param {object} req - Express Request Object
+ *
+ * @returns {promise} reject with error message; resolve true
+ */
+const revision = (req) => {
+  return new Promise((res, rej) => {
+    // no revision in the request
+    if (!req.params.hasOwnProperty('revision')) {
+      res(true);
+    }
+
+    if (req.params.revision === '' || !Number.isInteger(parseInt(req.params.revision, 10))) {
+      rej(config.content.messages.format.revision);
+    }
+
+    res(true);
+  });
+};
+
+/**
+ * Checking that what is in a content url matches what's in the system
+ * @param {object} req - HTTP Request
+ * @param {object} res - HTTP Response
+ * @param {object} next - Express callback
+ *
+ * @returns {function} express next function
+ */
+const url = (req, res, next) => {
```

Comment:
This is the pre-cursor to the route middleware. The middleware will give this function those parameters and only this function needs `next`. This function doesn't use `res`, but because this is middleware the order of these is already set.
The other functions do not need `next` as they are not triggering next but instead just taking their specific input and checking it for accuracy - it's this function which will decide what to do with the accuracy determination.